File: Semester4/Systems for Design and Implementation/mpp-proiect-java-raulfrandes/TriathlonModel/src/main/java/triathlon/model/validator/TrialValidator.java
package triathlon.model.validator;

import org.springframework.stereotype.Component;
import triathlon.model.Trial;

@Component
public class TrialValidator implements Validator<Trial>{
    @Override
    public void validate(Trial entity) throws ValidatorException {
        String errorMess = "";
        if (entity.getName().isEmpty()) {
            errorMess += "Invalid name - cannot be null";
        }
        if (entity.getDescription().isEmpty()) {
            errorMess += "Invalid name - cannot be null";
        }
        if (entity.getReferee() == null) {
            errorMess += "Invalid referee - cannot be null";
        }
        if (!errorMess.isEmpty()) {
            throw new ValidatorException(errorMess);
        }
    }
}

File: Semester4/Systems for Design and Implementation/mpp-proiect-csharp-raulfrandes/model/validator/TrialValidator.cs
namespace model.validator
{
    public class TrialValidator : IValidator<Trial>
    {
        public void Validate(Trial entity)
        {
            string errorMess = "";
            if (string.IsNullOrEmpty(entity.Name))
            {
                errorMess += "Invalid name - cannot be null\n";
            }

            if (string.IsNullOrEmpty(entity.Description))
            {
                errorMess += "Invalid description - cannot be null\n";
            }

            if (!string.IsNullOrEmpty(errorMess))
            {
                throw new ValidatorException(errorMess);
            }
        }
    }
}